src/query/plan/variable_start_planner.cpp

```diff
@@ -209,19 +209,43 @@ CartesianProduct<VaryMatchingStart> VaryMultiMatchingStarts(const std::vector<Ma
   return MakeCartesianProduct(std::move(variants));
 }
 
+CartesianProduct<VaryMatchingStart> VaryFilterMatchingStarts(const Matching &matching,
+                                                             const SymbolTable &symbol_table) {
+  auto i = 0;
+  for (const auto &filter : matching.filters) {
+    for ([[maybe_unused]] const auto &filter_matchings : filter.matchings) {
+      i += 1;
+    }
+  }
+
+  std::vector<VaryMatchingStart> variants;
+  variants.reserve(i);
+
+  for (const auto &filter : matching.filters) {
+    for (const auto &filter_matching : filter.matchings) {
+      variants.emplace_back(filter_matching, symbol_table);
+    }
+  }
+
```

Comment:
Yes it's a premature optimization :')

Reply:
I think we should not use `reserve` like this. It makes sense to reserve memory upfront if we are sure about the number of elements. Otherwise, we can just use `emplace_back`. What do you think?